proto/internal/temporal/server/api/persistence/v1/task_queues.proto

```diff
@@ -103,6 +104,30 @@ message VersioningData {
     repeated AssignmentRule assignment_rules = 2;
     // Unordered list of redirect rules. Also contains recently-deleted rules.
     repeated RedirectRule redirect_rules = 3;
+
+    // TaskQueueType based versioning info. This will be used for routing-related purposes
+    // for worker-versioning 3.
+    WorkflowTaskQueueBuildIDInfo workflow_task_queue_build_id_info = 4;
+    ActivityTaskQueueBuildIDInfo activity_task_queue_build_id_info = 5;
+    NexusTaskQueueBuildIDInfo nexus_task_queue_build_id_info = 6;
+
```

Comment:
The main reason why I went ahead with the separate types of deployment info for each task-queue type is because of my understanding that different task-queue types, of the same task-queue, can be present in different deployments. This made me realize that we do need a separate demarcation while dispatching tasks for the same task-queue

As always, the root task-queue partition shall own `UserData` but care will be taken in the implementation phase (especially in methods like `updateUserData` and more) to update the right `buildIDInfo`

Reply:
We need three fields, but I don't understand why we need three types. These can all be just `DeploymentInfo`, no?